main.cmake

```diff
@@ -44,6 +45,12 @@ option(MENDER_ARTIFACT_GZIP_COMPRESSION "Enable GZIP compression support when do
 option(MENDER_ARTIFACT_LZMA_COMPRESSION "Enable LZMA compression support when downloading and extracting Artifacts (Default: ON)" ON)
 option(MENDER_ARTIFACT_ZSTD_COMPRESSION "Enable Zstd compression support when downloading and extracting Artifacts (Default: ON)" ON)
 
+find_program(CCACHE ccache)
+if(CCACHE)
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE}")
```

Comment:
The cmake docs specify that `RULE_LAUNCH_COMPILE` [should not be used](https://cmake.org/cmake/help/latest/prop_tgt/RULE_LAUNCH_COMPILE.html#rule-launch-compile), and `CMAKE_CXX_COMPILER_LAUNCHER` should be used instead.